network/netconf/netconf_config.py

```diff
@@ -0,0 +1,229 @@
+    required: true
+
+
+requirements:
+  - "python >= 2.6"
+  - "ncclient"
+'''
+
+EXAMPLES = '''
+- name: set ntp server in the device
+  netconf_config:
+    host: 10.0.0.1
+    username: admin
+    password: admin
+    xml: |
+        <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
+            <system xmlns="urn:ietf:params:xml:ns:yang:ietf-system">
+                <ntp>
+                    <enabled>true</enabled>
+                    <server>
+                        <name>ntp1</name>
+                        <udp><address>127.0.0.1</address></udp>
+                    </server>
+                </ntp>
+            </system>
+        </config>
+
+- name: wipe ntp configuration
+  netconf_config:
+    host: 10.0.0.1
+    username: admin
+    password: admin
+    xml: |
+        <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
+            <system xmlns="urn:ietf:params:xml:ns:yang:ietf-system">
+                <ntp>
+                    <enabled>false</enabled>
+                    <server operation="remove">
+                        <name>ntp1</name>
+                    </server>
+                </ntp>
+            </system>
+        </config>
+
+'''
+
+RETURN = '''
+server_capabilities:
+    description: list of capabilities of the server
+    returned: success
+    type: list of strings
+    sample: ['urn:ietf:params:netconf:base:1.1','urn:ietf:params:netconf:capability:confirmed-commit:1.0','urn:ietf:params:netconf:capability:candidate:1.0']
+
+'''
+
+import xml.dom.minidom
+try:
+    import ncclient.manager
+    HAS_NCCLIENT = True
+except ImportError:
+    HAS_NCCLIENT = False
+
+
+import logging
+
+
+def netconf_edit_config(m, xml, commit, retkwargs):
+    if ":candidate" in m.server_capabilities:
+        datastore = 'candidate'
+    else:
+        datastore = 'running'
+    m.lock(target=datastore)
+    try:
+        m.discard_changes()
+        config_before = m.get_config(source=datastore)
+        m.edit_config(target=datastore, config=xml)
+        config_after = m.get_config(source=datastore)
+        changed = config_before.data_xml != config_after.data_xml
+        if changed and commit:
+            if ":confirmed-commit" in m.server_capabilities:
+                m.commit(confirmed=True)
+                m.commit()
+            else:
+                m.commit()
+        return changed
+    finally:
+        m.unlock(target=datastore)
+
+
+# ------------------------------------------------------------------- #
+# Main
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(type='str', required=True),
+            port=dict(type='int', default=830),
+            hostkey_verify=dict(type='bool', default=True),
+            username=dict(type='str', required=True, no_log=True),
+            password=dict(type='str', required=True, no_log=True),
+            operation=dict(choices=['edit_config'], default='edit_config'),
```

Comment:
Is this argument needed?   If there is only one choice, I don't think it needs to be exposed to the module playbook designer